### PR TITLE
fix: remove upper pin on vyper dependency

### PR DESCRIPTION
**Pull request by antazoey**
### What I did

trying to assist in envs where vyper is installed in 0.4

fixes: #114
fixes APE-1762

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

**Comment by antazoey** [comment]
After some discussion, going to look into actually ensuring we can flatten on 0.4...

- [ ] update with response or increased support

**Comment by mikeshultz** [comment]
> After some discussion, going to look into actually ensuring we can flatten on 0.4...

I haven't looked at the vyper changelog or anything, but my primary concern is vyper modules.  Right now imports are just treated as interfaces and transpiled accordingly.  Which is probably not what we want for modules.  But most of what I'm saying is guesswork.  Needs investigation.

File: setup.py [hunk 1]
```diff
@@ -62,7 +62,7 @@
         "ethpm-types",  # Use same version as eth-ape
         "tqdm",  # Use same version as eth-ape
         "vvm>=0.2.0,<0.3",
-        "vyper~=0.3.7",
+        "vyper>=0.3.7",  # Purposely no upper-pin.
```
[the remark this answers is not in the document]
Comment (antazoey) [reply setup.py:65]: nevermind! vyper 0.4 seems to be able to flatten 0.3 stuff so we good

**Comment by antazoey** [comment]
closing in favor of #118

**Pull request closed without merging**